src/Microsoft.Extensions.ML/PredictionEnginePool.cs

```diff
@@ -100,7 +101,7 @@ public ITransformer GetModel()
             //Here we are in the world of named models where the model hasn't been built yet.
             var options = _predictionEngineOptions.Create(modelName);
             var pool = new PoolLoader<TData, TPrediction>(_serviceProvider, options);
-            _namedPools.Add(modelName, pool);
+            _namedPools.TryAdd(modelName, pool);
```

Comment:
📝 This method should use `_namedPools.TryGetValue(modelName, out existingPool)` at the start of the method, rather than `ContainsKey` followed by indexing.

Reply:
Can you explain in detail the reason?

---
In reply to: [357425080](https://github.com/dotnet/machinelearning/pull/4570#discussion_r357425080) [](ancestors = 357425080)

Reply:
In this case, it's just more efficient. `ContainsKey` is doing a lookup, but the lookup value is ignored and then the indexer does exactly the same thing.

Reply:
Got it, thanks

---
In reply to: [357434839](https://github.com/dotnet/machinelearning/pull/4570#discussion_r357434839) [](ancestors = 357434839)